Example/AppDelegate.swift

```diff
@@ -1,23 +1,32 @@
 import Cocoa
 import Preferences
 
+extension NSToolbarItem.Identifier {
+	static var preferenceGeneral: NSToolbarItem.Identifier {
+		return NSToolbarItem.Identifier(rawValue: "preferenceGeneral")
+	}
+
+	static var preferenceAdvanced: NSToolbarItem.Identifier {
+		return NSToolbarItem.Identifier(rawValue: "preferenceAdvanced")
+	}
+}
+
 @NSApplicationMain
 final class AppDelegate: NSObject, NSApplicationDelegate {
 	@IBOutlet private var window: NSWindow!
 
-	let preferencesWindowController = PreferencesWindowController(
-		viewControllers: [
-			GeneralPreferenceViewController(),
-			AdvancedPreferenceViewController()
-		]
-	)
+	lazy var preferences: [Preferenceable] = [
+		GeneralPreferenceViewController(),
+		AdvancedPreferenceViewController()
+	]
+	lazy var preferencesWindowController: PreferencesWindowController = PreferencesWindowController(preferences: self.preferences, style: .segmentedControl, crossfadeTransitions: true)
 
 	func applicationWillFinishLaunching(_ notification: Notification) {
 		window.orderOut(self)
 	}
 
 	func applicationDidFinishLaunching(_ notification: Notification) {
-		preferencesWindowController.showWindow()
+		preferencesWindowController.showWindow(toolbarItemIdentifier: .preferenceAdvanced)
```

Comment:
I think this would read better as:

```swift
preferencesWindowController.showWindow(withPreference: .advanced)
```

Or something similar.

Reply:
Since `showWindow` would do the same as `change(to:)`, what do you think about the latest change to `showPreference`? I hope it brings across that it opens the window if needed, too.

Reply:
Yeah, that's good.